internal/api/v1/services/instance.go

```diff
@@ -103,77 +104,120 @@ func (s *Instance) GetInstance(ctx context.Context, ownerID, id uint) (*models.I
+	return nil
+}
+
 // provisionInstances provisions the job asynchronously
 func (s *Instance) provisionInstances(ctx context.Context, jobID uint, instances []infrastructure.InstanceRequest) {
-	// TODO: do something with the logs as now it makes the server logs messy
 	go func() {
-		fmt.Println("🚀 Starting async infrastructure creation...")
-
-		// Get the job name from the database
-		job, err := s.jobService.jobRepo.GetByID(ctx, 0, jobID) // ownerID 0 for now since we're not using it yet
+		// Get job name
+		job, err := s.jobService.jobRepo.GetByID(ctx, 0, jobID)
 		if err != nil {
-			fmt.Printf("❌ Failed to get job details: %v\n", err)
+			logger.Errorf("❌ Failed to get job details: %v", err)
 			return
 		}
 
-		// Create a JobRequest for provisioning
-		infra, err := infrastructure.NewInfrastructure(&infrastructure.InstancesRequest{
-			JobName:     job.Name,
-			Instances:   instances,
-			ProjectName: "example-project", // TODO: replace it with the actual project name in another PR
-			Action:      "create",
-			Provider:    instances[0].Provider, // Use the provider directly from the instance
-		})
+		// Create infrastructure client
+		infraReq := &infrastructure.InstancesRequest{
+			JobName:   job.Name,
+			Instances: instances,
+			Action:    "create",
+			Provider:  instances[0].Provider,
+		}
+
+		infra, err := infrastructure.NewInfrastructure(infraReq)
 		if err != nil {
-			fmt.Printf("❌ Failed to create infrastructure: %v\n", err)
+			logger.Errorf("❌ Failed to create infrastructure client: %v", err)
 			return
 		}
 
+		// Execute infrastructure creation
 		result, err := infra.Execute()
 		if err != nil {
-			// Check if error is due to resource not found
-			if strings.Contains(err.Error(), "404") &&
-				strings.Contains(err.Error(), "could not be found") {
-				fmt.Printf("⚠️ Warning: Some old resources were not found (already deleted)\n")
-			} else {
-				fmt.Printf("❌ Failed to execute infrastructure: %v\n", err)
-				return
-			}
+			logger.Errorf("❌ Failed to create infrastructure: %v", err)
+			return
 		}
 
-		// Update instance information in database
-		pInstances, ok := result.([]infrastructure.InstanceInfo)
+		// Type assert the result
+		pInstances, ok := result.([]types.InstanceInfo)
 		if !ok {
-			fmt.Printf("❌ Invalid result type: %T\n", result)
+			logger.Errorf("❌ Invalid result type: %T", result)
 			return
 		}
 
-		fmt.Printf("📝 Created instances: %+v\n", pInstances)
+		logger.Infof("📝 Created instances: %+v", pInstances)
 
+		// Update instances with IP and status
 		// Update instance information in database
 		ownerID := instances[0].OwnerID
 		for _, instance := range pInstances {
 			// Create update instance with only the fields we want to update
 			updateInstance := &models.Instance{
-				PublicIP: instance.IP,
+				PublicIP: instance.PublicIP,
 				Status:   models.InstanceStatusReady,
 			}
```

Comment:
Since other fields are not passed, they will be initialized with nil, zero, empty string, ... and the gorm, I think 90%, will update the fields with empty values. Please double check it and if that's the case, one solution would be to load the instance from database and update the fields we want and pass the entire object to the update function.

Reply:
BTW, as this is out of the scope of this PR, I'd suggest to have a followup issue for it.